src/convert.cpp

```diff
@@ -1517,7 +1517,11 @@ namespace {
             return false;
         }
         std::string outstr;
+#if defined(OS_SOLARIS)
+        const char* inptr = const_cast<char*>(str.c_str());
```

Comment:
Wouldn't it make sense to define `EXV_ICONV_CONST` to a different value on Solaris?